src/buf.ts

```diff
@@ -112,46 +112,46 @@ async function getDownloadURL(version: string, githubToken: string): Promise<str
     assetName = `buf-${platform}-${architecture}.tar.gz`
   }
   const octokit = new Octokit({ auth: githubToken });
-  const {data: releases} = await octokit.request(
-    'GET /repos/{owner}/{repo}/releases',
+  if (version === 'latest') {
+    const {data: releases} = await octokit.request(
+      'GET /repos/{owner}/{repo}/releases',
+      {
+        owner: 'bufbuild',
+        repo: 'buf',
+        per_page: 1,
+      }
+    );
+    for (const asset of releases[0].assets) {
+      if (assetName === asset.name) {
+        return asset.browser_download_url;
+      }
+    }
```

Comment:
This refactor means that if an asset was not found with the `/releases` endpoint above, we will still try the `/releases/tags` endpoint.

Previously, if an asset was not resolved from `latest`, we would `break` in the `switch`  return the following:

```
  return {
    message: `Unable to find Buf version "${version}" for platform "${platform}" and architecture "${architecture}".`
  };
```

With that said, if they configure the `latest` keyword in the action, then we'll always prioritize the latest release (via the `/releases` endpoint) and never resolve their specific tag named `latest`. It might be best to just return early here and prevent a second RPC that is destined to fail.

```suggestion
    }
    return {
      message: `Unable to find Buf version "${version}" for platform "${platform}" and architecture "${architecture}".`
    };
```

Reply:
That's a good call, big fan of the early return! :D